fenix/app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt

```diff
@@ -187,6 +187,10 @@ class ToolbarGestureHandler(
             is Destination.None -> return
         }
 
+        store.state.selectedTab ?.let {
+            val count = store.state.getNormalOrPrivateTabs(it.content.private).size
+            tabPreview.setTabCount(count)
+        }
```

Comment:
This logic would be moved to `TabPreview::onLayout` if following my suggestion above.